sources/niUI/src/GDRV_GLES2.cpp

```diff
@@ -4575,25 +4596,31 @@ struct cGLES2GraphicsDriver : public cIUnknownImpl<iGraphicsDriver>
         mCache.ShouldUpdate(&bShouldUpdate,eGLCache_RasterizerDepthOnly,
                             niFlagIs(matFlags,eMaterialFlags_DepthOnly));
         mCache.ShouldUpdate(&bShouldUpdate,eGLCache_RasterizerFlippedRT,isFlippedRT);
+        mCache.ShouldUpdate(&bShouldUpdate, eGLCache_ReversedCulling,
+                            isReversedCulling);
```

Comment:
Yeah...... the problem with this is that it breaks the batching, since now we have to look at every draw operations. The DrawOperationSet are usually sorted by state to minimise state changes, this could break that badly in pathological cases.

Reply:
Yeah I wasn't sure about this...But wouldn't a change on the Culling always be a change of state anyways? How can we do that all the objects with a ReverseCulling end up together so only one change of state is needed?

I am not sure if this helps BUT IF we don't care about animating the scale from positive to negative values (that's 99% of the cases?) we can check for the winding change only when bShouldUpdate is true and forget about having a cache for it.

Something like:
```
if (bShouldUpdate) {`
  // Check for determinant here
  // Call to Apply Raster States
}
```

Reply:
The thing is that if you dont reverse the culling you pretty much always have broken mesh or invisible mesh if you dont use DoubleSide, so it feels pointless? Can you think of a case where you'd do negative scaling and still want to keep the "wrong side" visible?

Reply:
Not really... :P